server/controllers/finance/cost_center_allocation_registry.js

```diff
@@ -0,0 +1,116 @@
+/**
+* Cost Center Allocation Keys
+*
+* This controller exposes an API to the client for reading allocation keys
+*/
+// const _ = require('lodash');
+const db = require('../../lib/db');
+const Stepdown = require('../../lib/stepdown');
+const fiscal = require('./fiscal');
+const ccAllocationKeys = require('./cost_center_allocation_bases');
+
+async function fetch(session, params) {
+  const defaultCurrentYear = new Date();
+  const enterpriseCurrencyId = session.enterprise.currency_id;
+
+  const periods = {
+    periodFrom : params.periodFrom || `${defaultCurrentYear.getFullYear()}01`,
+    periodTo : params.periodTo || `${defaultCurrentYear.getFullYear()}12`,
+  };
+
+  const range = await fiscal.getDateRangeFromPeriods(periods);
+  const query = 'CALL ComputeCostCenterAllocationByIndex(?, ?);';
+  let [feeCenters] = await db.exec(query, [range.dateFrom, range.dateTo]);
+
+  if (feeCenters.length) {
```

Comment:
I am confused by this.  What is it doing?  It looks like if a single result is returned, that means an weird happened?  Can we just throw the error and used a try/catch?

Reply:
It was a solution for the issue that we have got with `ComputeCostCenterAllocationByIndex` procedure, which thrown an error when there is not data to process for building the select query.

For the user, when he try to display the report when there is not data, the server returns an error message instead of just showing nothing as data in the report. So, to fix that I did these kind of checks (any optimal solution are welcome)

Reply:
Why isn't this a try/catch?  It doesn't look like an error is being thrown anywhere.